graphql/ownership.ts

```diff
@@ -30,6 +30,21 @@ export const isOwnedBy: { [Name in ResolverModelNames]?: (user: GraphQLUser, ent
         const instructor = await prisma.subcourse_instructors_student.findFirst({ where: { subcourseId: subcourse.id, studentId: user.studentId } });
         return !!instructor;
     },
+    Lecture: async (user, lecture) => {
+        const where = { appointmentId: lecture.id, status: AttendanceStatus.ACCEPTED };
+        switch (getUserType(user)) {
+            case 'student':
+                return (
+                    !!(await prisma.appointment_organizer.findFirst({ where: { ...where, studentId: user.studentId } })) ||
+                    !!(await prisma.appointment_participant_student.findFirst({ where: { ...where, studentId: user.studentId } }))
+                );
+            case 'pupil':
+                return !!(await prisma.appointment_participant_pupil.findFirst({ where: { ...where, pupilId: user.pupilId } }));
```

Comment:
Are Participants OWNER of an Appointment? Wouldn't it make more sense to grant that role only to the organizers, and then introduce another role for APPOINTMENT_PARTICIPANT (c.f. SUBCOURSE_PARTICIPANT)?

Reply:
I was hesitant of introducing new roles. Actually this would not work for our current workaround deriving participants from subcourse. Maybe we could lift the access restriction to not make it too complicated

Reply:
I don't understand "this would not work for our current workaround" ... Just move the code here into https://github.com/corona-school/backend/blob/master/graphql/authorizations.ts#L86 ? I think we need a new role here, and this will simplify writing further mutations for the entity

Reply:
I moved the code 👍🏻